File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/internal/storage/SharedPrefStore.java
package kin.sdk.internal.storage;


import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;


public class SharedPrefStore implements Store {

    private final SharedPreferences sharedPref;

    public SharedPrefStore(SharedPreferences sharedPref) {
        this.sharedPref = sharedPref;
    }

    @Override
    public void saveString(@NonNull String key, @NonNull String value) {
        sharedPref.edit()
                .putString(key, value)
                .apply();
    }

    @Override
    @Nullable
    public String getString(@NonNull String key) {
        return sharedPref.getString(key, null);
    }

    @Override
    public void clear(@NonNull String key) {
        sharedPref.edit().remove(key).apply();
    }
}

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/internal/storage/SharedPrefStore.kt
package kin.sdk.internal.storage

import android.content.SharedPreferences

internal class SharedPrefStore(private val sharedPref: SharedPreferences) : Store {

    override fun saveString(key: String, value: String) =
        sharedPref.edit()
            .putString(key, value)
            .apply()

    override fun getString(key: String): String? = sharedPref.getString(key, null)

    override fun clear(key: String) = sharedPref.edit().remove(key).apply()
}
